sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2216,21 +2216,25 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         numArgs: Int,
         u: UnresolvedFunction): Expression = {
       func match {
-        case owg: SupportsOrderingWithinGroup if u.isDistinct =>
+        case owg: InverseDistributionFunction if u.isDistinct =>
```

Comment:
Not related to this PR, but it seems unnecessary to disallow DISTINCT for functions supporting `WITHIN GROUP`. The Spark aggregation framework will handle the DISTINCT and the function implementation doesn't need to care about it at all.

Reply:
I don't know, but I think it's banned because the `percentile_cont(DISTINCT 0.5) WITHIN GROUP (ORDER BY v)` syntax seems incorrect. Without this `case` it will calculate 0.5 percentile over distinct `v` but IMO, it's confusing. I would rather expect `percentile_cont(0.5) WITHIN GROUP (ORDER BY DISTINCT v)` syntax for that.

Reply:
`ORDER BY DISTINCT v` is so confusing...

Reply:
The problem for `percentile_cont` is its percentile parameter muse be a constant, so it's meaning less to add DISTINCT.

Reply:
> The problem for percentile_cont is its percentile parameter muse be a constant, so it's meaning less to add DISTINCT.

But in practice `percentile_cont(DISTINCT 0.5) WITHIN GROUP (ORDER BY v)` will apply DISTINCT to `v`, not to constant parameter. And that behavior makes sense, but not with this syntax. Something like `percentile_cont(DISTINCT v, 0.5)` would be match better.

Now there are 3 extending class of `InverseDistributionFunction`:
1. `PercentileCont`
2. `PercentileDisc`
3. `Mode`

1 and 2 are strange with distinct and `mode` (returns the most frequent value) makes no sense with distinct.